test/SlowTests/Issues/RavenDB-20634.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using Newtonsoft.Json.Serialization;
+using Raven.Client.Documents.Queries;
+using Raven.Client.Json.Serialization.NewtonsoftJson;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_20634 : RavenTestBase
+{
+    public RavenDB_20634(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [Fact]
+    public void IdOfEmbeddedObjectIsNotWrappedIntoJavaScriptIdFunction()
+    {
+        using (var store = GetDocumentStore(options: new Options
+               {
+                   ModifyDocumentStore = ss =>
+                   {
+                       ss.Conventions.Serialization = new NewtonsoftJsonSerializationConventions
+                       {
+                           CustomizeJsonSerializer = s => { s.ContractResolver = new CamelCasePropertyNamesContractResolver(); }
+                       };
+                       ss.Conventions.PropertyNameConverter = mi => $"{Char.ToLower(mi.Name[0])}{mi.Name.Substring(1)}";
+                       ss.Conventions.ShouldApplyPropertyNameConverter = info => true;
+                   }
+               }))
+        {
+            using (var session = store.OpenSession())
+            {
+                var user = new MyUser { Id = "users/1", UserName = "john" };
+
+                session.Store(user);
+
+                var workspace = new Workspace { Id = "workspaces/1", Invites = new List<Invite> { new Invite { Id = "invites/1", Invitee = user.Id } } };
+
+                session.Store(workspace);
+
+                session.SaveChanges();
+            }
+ 
+            using (var session = store.OpenSession())
+            {
+                var query = (from workspace in session.Query<Workspace>()
```

Comment:
Can we assert `query.ToString()`? Just to make sure that we have there `x.id` (not `x.Id`)

Reply:
Done